Nos/Extensions/Font.swift

```diff
@@ -2,34 +2,37 @@ import SwiftUI
 
 // https://stackoverflow.com/a/74416073
 extension Font {
-    static func clarity(_ fontWeight: UIFont.Weight, textStyle: UIFont.TextStyle = .body) -> Font {
-        switch fontWeight {
-        case .regular:
-            clarityRegular(textStyle)
-        case .medium:
-            clarityMedium(textStyle)
-        case .semibold:
-            claritySemibold(textStyle)
-        case .bold:
-            clarityBold(textStyle)
-        default:
-            clarityRegular(textStyle)
-        }
+    
+    static func clarity(_ fontWeight: Font.Weight, textStyle: UIFont.TextStyle = .body) -> Font {
+        .custom(
+            fontWeight.clarityFontName,
+            size: UIFont.preferredFont(forTextStyle: textStyle).pointSize
+        )
     }
 
     static func clarityRegular(_ textStyle: UIFont.TextStyle) -> Font {
-        .custom("ClarityCity-Regular", size: UIFont.preferredFont(forTextStyle: textStyle).pointSize)
-    }
-
-    static func clarityMedium(_ textStyle: UIFont.TextStyle) -> Font {
-        .custom("ClarityCity-Medium", size: UIFont.preferredFont(forTextStyle: textStyle).pointSize)
+        .custom(
+            Font.Weight.regular.clarityFontName,
+            size: UIFont.preferredFont(forTextStyle: textStyle).pointSize
+        )
     }
-
-    static func claritySemibold(_ textStyle: UIFont.TextStyle) -> Font {
-        .custom("ClarityCity-SemiBold", size: UIFont.preferredFont(forTextStyle: textStyle).pointSize)
+    
+    static func clarityBold(baseSize size: CGFloat, relativeTo textStyle: Font.TextStyle) -> Font {
```

Comment:
What makes this different from the others such that it requires a baseSize in addition to a textStyle?

Reply:
The prior implementation doesn't require the caller to supply a base size because it uses this API to get a size instead:
`UIFont.preferredFont(forTextStyle: textStyle).pointSize`

This is supposed to scale with the user's font size, but it doesn't work as expected in the side menu. The problem is that this `Font` extension uses `Font.custom(_:size:)`, which the documentation says scales with the `body` text style. But if you pass in a font size that _already_ scales with the user's font size (from the `preferredFont...` API mentioned above), then you are scaling it _twice_. This results in truly gigantic text sizes in the "before" video.

Therefore, we need to use the newer API `Font.custom(_:size:relativeTo:)` AND pass a constant font size, since the API scales automatically. But where does that font size come from? I'll update it to use 20 as a default font size, and for this use-case, it works.